### PR TITLE
Clarifications to chunk format document.

### PR DESCRIPTION
**Pull request by nmoinvaz**
I have made some modifications for improved clarity. I am hoping I am understanding the format correctly.

### CODE REVIEW

File: README_CHUNK_FORMAT.rst [hunk 1]
```diff
@@ -103,40 +92,108 @@ All entries are little endian.
 :cbytes:
     (``uint32``) Compressed size of the buffer (including this header).
 
+:filter_codes:
+    (``uint8x6``) Filter code.
```
Comment (FrancescAlted) [inline README_CHUNK_FORMAT.rst:96]: uint8x6 -> uint8

File: README_CHUNK_FORMAT.rst [hunk 2]
```diff
@@ -103,40 +92,108 @@ All entries are little endian.
...
+    :``4``:
+        Truncate precision filter.
+
+    The filter pipeline has 6 reserved slots for the filters. They are applied sequentially to the chunk according
+    to their index in increasing order. The type of filter applied is specified by the `filter_code`. Each
+    `filter_code` has an associated field in `filter_meta` that can contain metadata about the filter.
+
+:filter_meta:
+
+    (``uint8x6``) Filter metadata.
```
Comment (FrancescAlted) [inline README_CHUNK_FORMAT.rst:117]: ditto

File: README_CHUNK_FORMAT.rst [hunk 3]
```diff
@@ -103,40 +92,108 @@ All entries are little endian.
...
-The blocks / splits section
----------------------------
+Each block is equal-sized as specified by the `blocksize` header field. The size of the last block that can be shorter
+or equal to the rest.
 
-After the header, there come the blocks / splits section.  Blocks are equal-sized parts of the chunk, except for the last block that can be shorter or equal than the rest.
+**Block starts**
 
-At the beginning of the blocks section, there come a list of `int32_t bstarts` to indicate where the different encoded blocks starts (counting from the end of this `bstarts` section)::
+The *block starts* section contains a list of offsets `int32_t bstarts` that indicate where each block starts in the
```
Comment (FrancescAlted) [inline README_CHUNK_FORMAT.rst:148]: For consistency with the rest of the document, I think it is better int32_t -> int32

File: README_CHUNK_FORMAT.rst [hunk 4]
```diff
@@ -103,40 +92,108 @@ All entries are little endian.
...
     | bstart0 | bstart1 |   ...  | bstartN |
     +=========+=========+========+=========+
 
-[**Only for C-Blosc2**] Next, it comes an optional dictionary section prepended with a `int32_t dsize`.  Dictionaries are small datasets that are known to be repeated a lot and can help to compress data in blocks/splits better::
+**Dictionary (optional)**
+
+*Only for C-Blosc2*
+
+Dictionaries are small datasets that are known to be repeated a lot and can help to compress data in blocks better.
+The dictionary section contains the size of the dictionary `int32_t dsize` followed by the dictionary data::
```
Comment (FrancescAlted) [inline README_CHUNK_FORMAT.rst:161]: int32_t -> int32

File: README_CHUNK_FORMAT.rst [hunk 5]
```diff
@@ -103,40 +92,108 @@ All entries are little endian.
...
+**Run-Length Encoding**
+
+*Only for C-Blosc2*
+
+The `csize` field of each compressed data stream can be used to support run-length encoding for blocks as follows:
 
-*Note*: all the integers are stored in little endian.
\ No newline at end of file
+    - When the most significant bit is *not* set, `csize` represents the size of the compressed
+      data stream that follows. (supported by C-Blosc1)
```
Comment (FrancescAlted) [inline README_CHUNK_FORMAT.rst:197]: "supported by" -> "as in".  This is the only meaning for `csize` in C-Blosc1, so I think this reads better this way?

File: README_CHUNK_FORMAT.rst [hunk 6]
```diff
@@ -103,40 +92,108 @@ All entries are little endian.
...
+    - When the most significant bit is set, the lowest significant byte of `csize` is used to fill the entire
+      block.
```
Comment (FrancescAlted) [inline README_CHUNK_FORMAT.rst:199]: Hmmm, you removed the examples.  Why so?

**Comment by nmoinvaz** [comment]
My idea with the compression stream terminology comes from zlib/deflate. As in each time the data goes through the compression codec it outputs a compressed stream of data. With deflate, it processes the data, creating blocks with literals/matches, and then at the end of the stream it finalizes it with a last block.

**Comment by nmoinvaz** [comment]
@FrancescAlted I have made the changes you have suggested. I also added to the _Compressed Data Streams_ section explaining what compressed data stream is.

File: README_CHUNK_FORMAT.rst [hunk 7]
```diff
@@ -103,40 +92,112 @@ All entries are little endian.
...
-    | dsize  | dictionary data |
-    +========+=================+
+Compressed data streams are the compressed set of bytes that are passed to codecs for decompression. Each stream is
+stored in the block with the size of the stream `int32 csize` followed by the bytes for the stream.
 
-Finally, it comes the actual list of compressed blocks / splits data streams.  It turns out that a block may optionally (see bit 4 in `flags` above) be further split in so-called splits which are the actual data streams that are transmitted to codecs for compression.  If a block is not split, then the split is equivalent to a whole block.  Before each split in the list, there is the compressed size of it, expressed as an `int32_t`::
+If bit 4 of the `flags` header field is *not* set, each block of is stored in a single data stream::
+
+    +========+========+
+    | csize0 | cdata0 |
```
Comment (FrancescAlted) [inline README_CHUNK_FORMAT.rst:175]: Why not use stream0 instead of cdata0?  It looks more consistent to me.

File: README_CHUNK_FORMAT.rst [hunk 8]
```diff
@@ -103,40 +92,112 @@ All entries are little endian.
...
+    +========+========+
+    | block0          |
+    +========+========+
+
+If bit 4 of the `flags` header is set, each block can be stored using multiple data streams::
 
     +========+========+========+========+========+========+========+
-    | csize0 | split0 | csize1 | split1 |   ...  | csizeN | splitN |
+    | csize0 | cdata0 | csize1 | cdata1 |   ...  | csizeN | cdataN |
```
Comment (FrancescAlted) [inline README_CHUNK_FORMAT.rst:183]: Ditto

**Comment by nmoinvaz** [comment]
I am taking a look at this again. What do you think of _compressed data blocks_ or _compressed data chunks_, instead of _compressed data streams_?

**Comment by nmoinvaz** [comment]
Actually it looks like you were using _compressed blocks_ in your document previously so maybe that makes the most sense.

**Comment by nmoinvaz** [comment]
@FrancescAlted I have made a few more modifications and renamed the sections to "streams" like you suggested, but also added another figure for `csize/cdata`.

**Comment by FrancescAlted** [comment]
Thanks!

**Comment by FrancescAlted** [comment]
Now we would need to update the [C-Blosc docs](https://github.com/Blosc/c-blosc/blob/master/README_CHUNK_FORMAT.rst) accordingly.

Also, there is reference inside the `blosc.c` and `blosc2.c` to splits.  That should be changed into streams.

**Pull request merged**